### DISCUSSION TITLE
Optimizing constant branches

### DISCUSSION THREAD

**Discussion by YulienPohl** [hneemann/Digital#660] [Ideas]
The current speed of the simulator is already fast but could be improved when constants are used.
I know not driven constant signals are optimized however not branches that are driven but not used.

For example
![constant_value](https://user-images.githubusercontent.com/37116826/108606059-206b1f00-73b8-11eb-9e52-2556aeb6a52c.png)
is optimized

![constant_branch](https://user-images.githubusercontent.com/37116826/108606066-28c35a00-73b8-11eb-957e-7e129d4a85a2.png)
is not despite the constant multiplexer, and gate, etc.

**Comment by hneemann** [comment]
Such optimization would only make a difference when such circuits are built, and I don't think that happens often enough to be worth the effort.

**Reply by YulienPohl** [reply]
You're right, it is only necessary when complex generic circuits are controlled with constant gates.
